internal/mocks/mock_tracing_server.go

```diff
@@ -0,0 +1,50 @@
+package mocks
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"sync"
+
+	otlpcollector "go.opentelemetry.io/proto/otlp/collector/trace/v1"
+	"google.golang.org/grpc"
+)
+
+type mockTracingServer struct {
+	otlpcollector.UnimplementedTraceServiceServer
+	exportCount int
+	serviceMu   sync.Mutex
+}
+
+var _ otlpcollector.TraceServiceServer = (*mockTracingServer)(nil)
+
+func (s *mockTracingServer) Export(context.Context, *otlpcollector.ExportTraceServiceRequest) (*otlpcollector.ExportTraceServiceResponse, error) {
+	s.serviceMu.Lock()
+	defer s.serviceMu.Unlock()
+	s.exportCount++
+	return &otlpcollector.ExportTraceServiceResponse{}, nil
+}
+
+func NewMockTracingServer(port int) (*mockTracingServer, error) {
+	mockServer := &mockTracingServer{exportCount: 0}
+
+	go func() {
+		lis, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
```

Comment:
I'm still looking into a way to mock this with `mockgen` and not having to have this, but in the meantime the https://pkg.go.dev/google.golang.org/grpc/test/bufconn package is a pretty standard package to use to test grpc services. You can avoid network and instead use a buffered loop connection. We should change this to the `bufconn` package, because then you won't get those firewall pop-ups 😄

Reply:
I gave it a try and the test fails here https://github.com/openfga/openfga/blob/df6f05a3d2ac1d8472079057d4b73c3e337f59ce/pkg/telemetry/tracing.go#L68C1-L75

with `panic: failed to establish a connection with the otlp exporter: context deadline exceeded`